src/box/alter.cc

```diff
@@ -1988,6 +1988,46 @@ space_check_truncate(struct space *space)
 	return 0;
 }
 
+/**
+ * Check whether @a old_space holders prohibit alter to @a new_space_def.
+ * For example if the space becomes temporary, there can be foreign keys
+ * from non-temporary space, so this alter must not be allowed.
+ * Return 0 if allowed, or -1 if not allowed (diag is set).
+ */
+static int
+space_check_alter(struct space *old_space, struct space_def *new_space_def)
+{
```

Comment:
Nit: Let's add an assertion here checking that the space's `is_local` flag didn't chnage:

```C
assert(old_space->def->group_id == new_space_def->group_id);
```

Reply:
added.